packages/app/src/debug/DebugPageHeader.vue

```diff
@@ -144,6 +152,16 @@ const props = defineProps<{
 
 const debug = computed(() => props.gql)
 
+const ICON_MAP = {
+  PASSED: { textColor: 'text-jade-400' },
+  FAILED: { textColor: 'text-red-400' },
+  CANCELLED: { textColor: 'text-gray-500' },
+  ERRORED: { textColor: 'text-orange-400' },
+  RUNNING: { textColor: 'text-indigo-500' },
+}
+
+const runNumberColor = computed(() => ICON_MAP[String(props.gql.status)]['textColor'])
+
```

Comment:
Suggestion - to make this a little more typed you could do something like this

```suggestion
const ICON_MAP: { [key in CloudRunStatus]?: {textColor: string} } = {
  PASSED: { textColor: 'text-jade-400' },
  FAILED: { textColor: 'text-red-400' },
  CANCELLED: { textColor: 'text-gray-500' },
  ERRORED: { textColor: 'text-orange-400' },
  RUNNING: { textColor: 'text-indigo-500' },
}

const runNumberColor = computed(() => {
  if (props.gql.status) {
    const icon = ICON_MAP[props.gql.status]

    return icon?.textColor
  }

  return ''
})
```

The only issue is that you'd then have to cast it to `any` when passing it to `SolidStatusIcon` because it won't accept `string`.

```jsx
<SolidStatusIcon
    size="16"
    :status="debug.status.toLowerCase() as any"
/>
```

up to you

Reply:
I tried doing

```
const ICON_MAP: Record<CloudRunStatus, { textColor: string }> = {
  PASSED: { textColor: 'text-jade-400' },
  FAILED: { textColor: 'text-red-400' },
  CANCELLED: { textColor: 'text-gray-500' },
  ERRORED: { textColor: 'text-orange-400' },
  RUNNING: { textColor: 'text-indigo-500' },
} as const
```

Error...

```
Type '{ readonly PASSED: { readonly textColor: "text-jade-400"; }; readonly FAILED: { readonly textColor: "text-red-400"; }; readonly CANCELLED: { readonly textColor: "text-gray-500"; }; readonly ERRORED: { readonly textColor: "text-orange-400"; }; readonly RUNNING: { readonly textColor: "text-indigo-500"; }; }' is missing the following properties from type 'Record<CloudRunStatus, { textColor: string; }>': NOTESTS, OVERLIMIT, TIMEDOUTts(2739)
```

Are our types telling us something? What do we do for `NOTESTS, OVERLIMIT, TIMEDOUT`? Should it be an empty string? In this case would we just add those definitions?

If we do, this becomes

```ts
import type { CloudRunStatus  } from '../generated/graphql'

const ICON_MAP: Record<CloudRunStatus, { textColor: string }> = {
  PASSED: { textColor: 'text-jade-400' },
  FAILED: { textColor: 'text-red-400' },
  CANCELLED: { textColor: 'text-gray-500' },
  ERRORED: { textColor: 'text-orange-400' },
  RUNNING: { textColor: 'text-indigo-500' },
  NOTESTS: { textColor: 'text-indigo-500' },
  OVERLIMIT: { textColor: 'text-indigo-500' },
  TIMEDOUT: { textColor: 'text-indigo-500' },
} as const

const runNumberColor = computed(() => {
  if (!props.gql.status) {
    return ''
  }
  return ICON_MAP[props.gql.status].textColor
})
```

Which seems to be nice and  type safe, and represent all states.

Reply:
I believe the suggestions above are definitely the better approach here. Type safety is important and adding this would improve that!